hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/map/MapIndexScanPMigrationStressTest.java

```diff
@@ -0,0 +1,192 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.connector.map;
+
+import com.hazelcast.config.IndexConfig;
+import com.hazelcast.config.IndexType;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.jet.SimpleTestInClusterSupport;
+import com.hazelcast.jet.sql.SqlTestSupport;
+import com.hazelcast.map.IMap;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.hazelcast.jet.sql.SqlTestSupport.assertRowsOrdered;
+
+public class MapIndexScanPMigrationStressTest extends SimpleTestInClusterSupport {
+    static final int ITEM_COUNT = 750_000;
+    static final int ITEM_TO_LOOKUP = 250_000;
+    static final int COUNT_DIVIDER = 3;
+    static final int MEMBERS_COUNT = 3;
+    static final String MAP_NAME = "map";
+
+    private IMap<Integer, Integer> map;
+
+    @BeforeClass
+    public static void setUpClass() {
+        initializeExceptLast(MEMBERS_COUNT, smallInstanceConfig());
+    }
+
+    @Before
+    public void before() {
+        map = instance().getMap(MAP_NAME);
+    }
+
+    @Test
+    @Ignore // TODO: [sasha] un-ignore after IMDG engine removal
+    public void stressTestPointLookup() {
+        List<SqlTestSupport.Row> expected = new ArrayList<>();
+        for (int i = 0; i <= ITEM_COUNT; i++) {
+            map.put(i, i);
+        }
+        expected.add(new SqlTestSupport.Row(ITEM_TO_LOOKUP, ITEM_TO_LOOKUP));
+
+        IndexConfig indexConfig = new IndexConfig(IndexType.HASH, "this").setName(randomName());
+        map.addIndex(indexConfig);
+
+        MutatorThread mutator = new MutatorThread(instances(), instances().length - 1, 500L);
+        mutator.start();
+
+        assertRowsOrdered("SELECT * FROM " + MAP_NAME + " WHERE this=" + ITEM_TO_LOOKUP, expected);
+
+        try {
+            mutator.join();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    @Ignore // TODO: [sasha] un-ignore after IMDG engine removal
+    public void stressTestSameOrder() {
+        List<SqlTestSupport.Row> expected = new ArrayList<>();
+        for (int i = 0; i <= ITEM_COUNT; i++) {
+            map.put(i, i);
+            expected.add(new SqlTestSupport.Row(ITEM_COUNT - i, ITEM_COUNT - i));
+        }
+
+        IndexConfig indexConfig = new IndexConfig(IndexType.SORTED, "this").setName(randomName());
+        map.addIndex(indexConfig);
+
+        MutatorThread mutator = new MutatorThread(instances(), instances().length - 1);
+        mutator.start();
+
+        assertRowsOrdered("SELECT * FROM " + MAP_NAME + " ORDER BY this DESC", expected);
```

Comment:
With single query, we are rather stress testing cluster membership changes, not the correctness of index based queries.